Tasks/JenkinsQueueJob/util.ts

```diff
@@ -158,4 +210,113 @@ function submitJob(initialPostData: any, taskOptions: TaskOptions): Q.Promise<st
     }).auth(taskOptions.username, taskOptions.password, true);
 
     return defer.promise;
-}
\ No newline at end of file
+}
+
+function getCrumb(taskOptions: TaskOptions): Q.Promise<string> {
+    let defer: Q.Deferred<string> = Q.defer<string>();
+    let crumbRequestUrl: string = addUrlSegment(taskOptions.serverEndpointUrl, '/crumbIssuer/api/xml?xpath=concat(//crumbRequestField,%22:%22,//crumb)');
+    tl.debug('crumbRequestUrl: ' + crumbRequestUrl);
+    request.get({ url: crumbRequestUrl }, function (err, httpResponse, body) {
+        if (err) {
+            if (err.code == 'ECONNRESET') {
+                tl.debug(err);
+                defer.resolve(null);
+            } else {
+                defer.reject(err);
+            }
+        } else if (httpResponse.statusCode == 404) {
+            tl.debug('crumb not found');
+            taskOptions.crumb = taskOptions.NO_CRUMB;
+            defer.resolve(taskOptions.NO_CRUMB);
+        } else if (httpResponse.statusCode != 200) {
+            failReturnCode(httpResponse, 'crumb request failed.');
+            defer.reject(getFullErrorMessage(httpResponse, 'Crumb request failed.'));
+        } else {
+            taskOptions.crumb = body.slice(body.indexOf(':') + 1);
+            tl.debug('crumb: ' + taskOptions.crumb);
+            defer.resolve(taskOptions.crumb);
+        }
+    }).auth(taskOptions.username, taskOptions.password, true);
+    return defer.promise;
+}
+
+/**
+ * Supported parameter types: boolean, string, choice, password
+ * 
+ * - If a parameter is not defined by Jenkins it is fine to pass it anyway
+ * - Anything passed to a boolean parameter other than 'true' (case insenstive) becomes false.
+ * - Invalid choice parameters result in a 500 response.
+ * 
+ */
+function parseJobParameters(jobParameters: string[]) {
+    var formData = {};
+    // first add built in parameters
+    allTeamBuildVariables.forEach(variableName => {
+        let paramValue = tl.getVariable(variableName);
+        if (paramValue) {
+            let paramName = '_team-build_' + variableName;
+            formData[paramName] = paramValue;
+        }
+    });
+
+    // then add user parameters
+    for (var i = 0; i < jobParameters.length; i++) {
+        var paramLine = jobParameters[i].trim();
+        var splitIndex = paramLine.indexOf('=');
+        if (splitIndex <= 0) { // either no paramValue (-1), or no paramName (0)
+            throw 'Job parameters should be specified as "parameterName=parameterValue" with one name, value pair per line. Invalid parameter line: ' + jobParameters[i];
+        }
+        var paramName = paramLine.substr(0, splitIndex).trim();
+        var paramValue = paramLine.slice(splitIndex + 1).trim();
+        formData[paramName] = paramValue;
+    }
+    return formData;
+}
+
+//https://www.visualstudio.com/docs/build/define/variables
+var allTeamBuildVariables: string[] = [
```

Comment:
To my knowledge, there is no mechanism in place right now to get all of the variables.  We'll need to update this if there are new variables that come along we need to pass.